containers/osbuild-composer/entrypoint.py

```diff
@@ -0,0 +1,136 @@
+    os.chdir("/usr/libexec/osbuild-composer")
+
+
+#
+# Prepare Sockets
+#
+# Prepare all the API sockets that osbuild-composer expectes, and make sure to
+# pass them according to the systemd socket-activation API.
+#
+# Note that we rely on this being called early, so we get the correct FD
+# numbers assigned. We need FD-#3 onwards for compatibility with socket
+# activation (because python `subprocess.Popen` does not support renumbering
+# the sockets we pass down).
+#
+def prepare_sockets():
+    index = 3
+    sockets = []
+    fds = []
+    names = []
+
+    # osbuild-composer.socket:0
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind("/run/weldr/api.socket")
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-composer.socket")
+
+    assert(sock.fileno() == index)
+    index += 1
+
+    # osbuild-composer.socket:1
+    sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    sock.bind("/run/osbuild-composer/job.socket")
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-composer.socket")
+
+    assert(sock.fileno() == index)
+    index += 1
+
+    # osbuild-composer-api.socket:2
+    sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
+    sock.bind(("::", 443))
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-composer-api.socket")
+
+    assert(sock.fileno() == index)
+    index += 1
+
+    # osbuild-remote-worker.socket:3
+    sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    sock.setsockopt(socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
+    sock.bind(("::", 8700))
+    sock.listen(256)
+    sockets.append(sock)
+    fds.append(sock.fileno())
+    names.append("osbuild-remote-worker.socket")
+
+    assert(sock.fileno() == index)
+    index += 1
+
+    # Prepare FD environment for the child process.
+    os.environ["LISTEN_FDS"] = str(len(sockets))
+    os.environ["LISTEN_FDNAMES"] = ":".join(names)
+
+    return (sockets, fds)
+
+
+#
+# Execute osbuild-composer
+#
+# Synchronously execute Composer and keep all the temporary resources alive
+# until composer exits. On interrupts, terminate Composer and return control.
+#
+def run_composer(fds):
+    cmd = [
+        "/usr/libexec/osbuild-composer/osbuild-composer",
+        "-v",
+    ]
+
+    preexec_setenv = lambda: os.putenv("LISTEN_PID", str(os.getpid()))
+
+    try:
+        proc = subprocess.Popen(
+            cmd,
+            stdin=subprocess.DEVNULL,
+            stderr=subprocess.STDOUT,
+            pass_fds=fds,
+            preexec_fn=preexec_setenv,
+        )
+        return proc.wait()
+    except:
+        proc.terminate()
+        raise
```

Comment:
I'm not sure this will be a common use case, but I'm always getting a 23-line stack trace when stopping the container with `Ctrl-C`.

We could not re-raise the exception on `KeyboardInterrupt` and `SIGTERM`. Can be in a follow-up though.